### PR TITLE
failpoints: setup for ci only once

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

## What have you changed? (mandatory)
Call `test_util::setup_for_ci()` only once before any other tests.

## What are the type of changes? (mandatory)
- Bugfix (a change which fixes an issue)

## How has this PR been tested? (mandatory)
```
$ env CI=ok LOG_FILE=foo cargo test --test failpoints --features failpoints
test result: ok. 46 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

### CODE REVIEW

File: tests/failpoints/mod.rs [hunk 1]
```diff
@@ -10,9 +10,13 @@ extern crate slog_global;
...
 
 fn setup<'a>() -> fail::FailScenario<'a> {
-    let guard = fail::FailScenario::setup();
+    fail::FailScenario::setup()
+}
+
+// The prefix "_" here is to guarantee running this case first.
+#[test]
+fn _ci_setup() {
     test_util::setup_for_ci();
```
Comment (overvenus) [inline tests/failpoints/mod.rs]: How about `Once.call_once(test_util::setup_for_ci)`?

Also, could you call `setup` twice in `test_setup`?

Comment (youjiali1995) [reply tests/failpoints/mod.rs]: Did you mean to put `Once.call_once(test_util::setup_for_ci)` to `setup()` and no need to have `_ci_setup()`?

File: tests/failpoints/mod.rs [hunk 2]
```diff
@@ -9,10 +9,13 @@ extern crate slog_global;
...
+use std::sync::Once;
+
+static INIT: Once = Once::new();
+
 fn setup<'a>() -> fail::FailScenario<'a> {
-    let guard = fail::FailScenario::setup();
-    test_util::setup_for_ci();
-    guard
+    INIT.call_once(test_util::setup_for_ci);
+    fail::FailScenario::setup()
```
Comment (breeswish) [inline tests/failpoints/mod.rs]: why fail scenarios are also set up multiple times?

Comment (youjiali1995) [reply tests/failpoints/mod.rs]: https://docs.rs/fail/0.3.0/fail/#usage-in-tests

File: tests/failpoints/mod.rs [hunk 3]
```diff
@@ -9,10 +9,13 @@ extern crate slog_global;
 
 mod cases;
 
+use std::sync::Once;
+
+static INIT: Once = Once::new();
+
 fn setup<'a>() -> fail::FailScenario<'a> {
```
Comment (overvenus) [inline tests/failpoints/mod.rs]: Could you call `let _ = setup();` between line 23 and 24? So we can test whether multiple setups lead to panic.

Comment (youjiali1995) [reply tests/failpoints/mod.rs]: It will cause deadlock because `fail::FailScenario::setup()` acquires a lock and releases it when drops.

Comment (youjiali1995) [reply tests/failpoints/mod.rs]: Oh, sorry. It will be dropped before the next `setup()`.

**Pull request merged**